clients/src/main/java/org/apache/kafka/common/record/CompressionType.java

```diff
@@ -32,6 +118,10 @@
         this.rate = rate;
     }
 
+    public abstract DataOutputStream wrapForOutput(ByteBufferOutputStream buffer, byte messageVersion, int bufferSize);
+
+    public abstract DataInputStream wrapForInput(ByteBufferInputStream buffer, byte messageVersion);
```

Comment:
Should this return a `DataInputStream` or simply an `InputStream`? Seems like wrapping into a `DataInputStream` is not relevant for this class and can be done by the caller. Same for the output case.